### ISSUE TITLE
Rule Change: add siblingsOnly option to no-duplicate-headings rule

### ISSUE THREAD

**Issue by xbinaryx** [eslint/markdown#390]
### What rule do you want to change?

no-duplicate-headings

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```markdown
# Change log

## 1.0.0

### Features
### Bug Fixes

## 2.0.0

### Features
### Bug Fixes
```

### What does the rule currently do for this code?

Currently, the no-duplicate-headings rule would flag all duplicate headings at the same level, regardless of their parent heading context. For the example above, it would report:

Duplicate heading "Features" found
Duplicate heading "Bug Fixes" found

### What will the rule do after it's changed?

With the new siblingsOnly: true option enabled,  the rule will only check for duplicates among headings that share the same immediate parent heading.

For the example changelog:

- No violations will be reported
- Each "Features" heading is allowed because they have different parent headings ("1.0.0" vs "2.0.0")
- Similarly, the "Bug Fixes" headings are allowed

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

Prior Art: [MD024](https://github.com/DavidAnson/markdownlint/blob/main/doc/md024.md)

**Comment by nzakas** [comment]
Sounds good to me. 👍
